### ISSUE TITLE
Tie up loose ends on ClusterRole Validation

### ISSUE THREAD

**Issue by xtreme-sameer-vohra** [knative-sandbox/discovery#175]
Blocked on https://github.com/knative-sandbox/discovery/pull/145 (**this is no longer blocked**)

- [ ] ~~Validation Webhook to test RoleRef (validate if this is required)~~
- [x] Leverage OpenAPI schema for basic validation
- [x] Increase test coverage for methods introduced into reconciler in https://github.com/knative-sandbox/discovery/pull/145 (Check if cache is being polluted hence new test cases are not working, 
 Scott mentioned looking at doing a deep copy instead in the test setup code) will be done as part of [ref](https://github.com/knative-sandbox/discovery/issues/182)
- [x] Document the new RoleRef in the CDT in the README
- [x] Update the README to show the updated CDT Status
- [x] E2E test that uses a RoleRef
- [x] Document using schema tool to update the CRD
- [x] ~~Show a better error if cluster role is not defined - right now it just shows observed gen failure~~
- [x] Open separate issue for how ClusterRole permissions were checked against ducks when the plural form the duck isn't just appending a "s" at the end ([link](https://github.com/knative-sandbox/discovery/issues/182))

**Comment by xtreme-sameer-vohra** [comment]
Knative's schema validation flags can't be applied to vendored code, so unable to add validation to ensure a ClusterRole's Name is populated.

**Comment by xtreme-sameer-vohra** [comment]
If the CDT's configured clusterRole is not found, an error is populated in the events:
k describe cducks/some-cdt-configured-with-non-existent-cluster-role-ref

```
.
.
Events:
  Type     Reason         Age                 From                        Message
  ----     ------         ----                ----                        -------
  Warning  InternalError  29s (x17 over 88s)  clusterducktype-controller  clusterroles.rbac.authorization.k8s.io "non-existent-cluster-role" not found
```

**Comment by xtreme-sameer-vohra** [comment]
Closing this AFAIK all the action items above have been addressed.
